Assets/Scripts/HUD/TimerCountdown.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.UI;
+
+public class TimerCountdown : MonoBehaviour
+{
+    public TextMeshProUGUI countdownTextDisplay;
+    public GameObject endMissionText;
+    public int secondsLeft = 5 * 60;
+    private bool takingAway = false;
+
+    void Start()
+    {
+        if (countdownTextDisplay != null) countdownTextDisplay.text = CountdownString();
+    }
+
+    void Update()
+    {
+        if (!takingAway && secondsLeft > 0)
+        {
+            StartCoroutine(TimerTake());
+        }
+
+        if (secondsLeft <= 0)
```

Comment:
For now it's not a big deal, but stuff like that try to do with events like OnCountdownEnd or something like that, or you're just running that line about 100 times per second which is not great. (Or just take it out of the update)